orchestrator/careplanservice/handle_getpatient.go

```diff
@@ -0,0 +1,146 @@
+package careplanservice
+
+import (
+	"context"
+	"fmt"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+	"net/url"
+)
+
+// handleGetPatient fetches the requested Patient and validates if the requester has access to the resource (is a participant of one of the CareTeams associated with the patient)
+// if the requester is valid, return the Patient, else return an error
+// Pass in a pointer to a fhirclient.Headers object to get the headers from the fhir client request
+func (s *Service) handleGetPatient(ctx context.Context, id string, headers *fhirclient.Headers) (*fhir.Patient, error) {
+	// Verify requester is authenticated
+	_, err := auth.PrincipalFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var patient fhir.Patient
+	err = s.fhirClient.Read("Patient/"+id, &patient, fhirclient.ResponseHeaders(headers))
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the CarePlan for which the Patient is the subject, get the CareTeams associated with the CarePlan
+	// The search for CarePlan already checks auth for CareTeam, so if we get back a valid CarePlan, we can assume the user has access to the patient
+	// If the user is not part of any of these CareTeams, the search for CarePlan will return an error
+	bundle, err := s.handleSearchCarePlan(ctx, url.Values{"subject-identifier": []string{patientBSNFromIdentifierList(patient.Identifier)}}, headers)
```

Comment:
I see creation of the custom search parameter `subject-identifier` often failing, we need to check if this works on Azure FHIR.

Reply:
I would de-prioritize this one. I don't think we need this custom search-param as we now use 'normal' local references to a patient-instance. The (CarePlan) 'subject' or 'patient' search-param is a standard one for HAPI and Azure FHIR.

Reply:
Ok, noted. I will switch to searching on Subject or Patient (need to investigate which we use)

Reply:
> I would de-prioritize this one. I don't think we need this custom search-param as we now use 'normal' local references to a patient-instance. The (CarePlan) 'subject' or 'patient' search-param is a standard one for HAPI and Azure FHIR.

Yes, but...: we use a logical identifier in `CarePlan.for` with BSN, so we still need to search for the patient using logical identifier. I can't remember why @JorisHeadease decided to make a custom search parameter.. 🤔